apple/Common/SHPathUtils.h

```diff
@@ -0,0 +1,17 @@
+//
+//  SHPathUtils.h
+//  Sourcehold
+//
+//  Created by apodrugin on 08.02.2021.
+//
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+bool SHGetUserDocumentsDirectoryPath(char *buffer, unsigned long bufferSize);
+bool SHGetMainBundleResourcesDirectoryPath(char *buffer, unsigned long bufferSize);
+
```

Comment:
This is naming convention from Apple world (you can see examples in CoreFoundation, CFArrayGetValues for instance). These methods obtain some value (i.e. 'get') and not just 'set' some pointers value. I think that this naming better reflects the essence of what is happening.

Reply:
Fair enough, however I disagree since in the end this is a C interface, not Obj-C.
I checked the CoreFoundation documention and you are indeed right, so I don't blame you.

When I read 'Get', I usally think of getting a pointer or a reference to some object.
I don't have to worry about the memory of the object, because it is managed by the interface.

Reply:
What do you think about SHCopyUserDocumentsDirectoryPathToBuffer and SHCopyMainBundleResourcesDirectoryPathToBuffer?

Reply:
I just figured that 'Set' is also a bit misleading, I admit that. It implies setting something to the value of buffer.
How about:
```C++
bool SHQueryUserDocumentsDirectoryPath(char* out, unsigned long out_size)
```

Reply:
Sounds good 👍

Reply:
Thank you for accepting the convention.